jupyter-org-extensions.el

```diff
@@ -0,0 +1,404 @@
+    (jupyter-org-insert-src-block below)
+    (delete-char 1)
+    (insert code)
+    ;; move to the end of the last line of the cloned block
+    (forward-line -1)
+    (end-of-line)))
+
+;;;###autoload
+(defun jupyter-org-merge-blocks ()
+  "Merge the current block with the next block."
+  (interactive)
+  (let ((current-src-block (org-element-context))
+        (next-src-block (save-excursion
+                          (org-babel-next-src-block)
+                          (org-element-context))))
+    (unless (eq (org-element-type current-src-block) 'src-block)
+      (error "Not in a source block"))
+    (let ((merged-code (concat (org-element-property :value current-src-block)
+                               (org-element-property :value next-src-block)))
+          (lang (org-element-property :language current-src-block))
+          (switches (or (org-element-property :switches current-src-block) ""))
+          (parameters (or (org-element-property :parameters current-src-block) "")))
+      ;; Remove source blocks
+      (mapc (lambda (src-block)
+              (goto-char (org-element-property :begin src-block))
+              (org-babel-remove-result)
+              (setf (buffer-substring (org-element-property :begin src-block)
+                                      (org-element-property :end src-block))
+                    ""))
+            (list next-src-block current-src-block))
+      ;; Now create the merged block, point is where the current block was
+      (insert
+       (org-element-interpret-data
+        (org-element-put-property
+         (jupyter-org-src-block lang parameters merged-code switches)
+         :post-blank 1)))
+      ;; (insert (format "#+BEGIN_SRC %s %s %s\n%s#+END_SRC\n\n"
+      ;;                 lang switches parameters merged-code))
+      (forward-line -3)
+      (end-of-line))))
+
+;;;###autoload
+(defun jupyter-org-move-src-block (&optional below)
+  "Move source block before of after another.
+
+If BELOW is non-nil, move the block down, otherwise move it up."
+  (interactive)
+  ;;skip all this if there are no previous or next source block
+  (when (condition-case nil
+            (save-excursion
+              (if below
+                  (org-babel-next-src-block)
+                (org-babel-previous-src-block)))
+          (error nil))
+    (let* ((src (org-element-context))
+           (results-start (org-babel-where-is-src-block-result))
+           (results-end
+            (when results-start
+              (save-excursion
+                (goto-char results-start)
+                (goto-char (org-babel-result-end))
+                ;; if line is empty, take that empty line by moving down
+                (when (looking-at-p "[[:space:]]*$")
+                  (forward-line 1))
+                (point)))))
+      ;; kill from the start of the source block to the end of the results
+      ;; ... if there are no results, to the end of the source block
+      (kill-region
```

Comment:
I think it would be good to use `delete-and-extract-region` here. Is there a reason to use `kill-region`?

Reply:
Good point. I've switched to `delete-and-extract-region`.
I've also simplified this function (i.e. `jupyter-org-move-src-block`).